src/main/scala/com/gu/anghammarad/models/Serialization.scala

```diff
@@ -1,38 +1,74 @@
 package com.gu.anghammarad.models
 
+import com.gu.anghammarad.Enrichments._
 import io.circe.Decoder.Result
 import io.circe._
-import io.circe.generic.semiauto._
 import io.circe.parser._
-import io.circe.syntax._
 
 import scala.util.{Failure, Success, Try}
-import com.gu.anghammarad.Enrichments._
 
 
 object Serialization {
-  def parseMapping(jsonStr: String): Try[Mapping] = {
-    val tryJson = parse(jsonStr).fold(
+
+  def parseAllMappings(jsonStr: String): Try[List[Mapping]] = {
+    val allMappings = for {
+      json <- parse(jsonStr)
+      rawMappings <- json.hcursor.downField("mappings").as[List[Json]]
+      mappings <- rawMappings.traverseE(parseMapping)
+    } yield mappings
+
+    allMappings.fold(
       err => Failure(err),
-      json => Success(json)
+      suc => Success(suc)
     )
+  }
 
+  private[models] def parseMapping(json: Json): Result[Mapping] = {
+    val hCursor = json.hcursor
+    for {
+      rawTargets <- hCursor.downField("target").as[Json]
+      rawContacts <- hCursor.downField("contacts").as[Json]
+      targets = parseTargets(rawTargets)
+      contacts = parseContacts(rawContacts)
+    } yield Mapping(targets, contacts)
+  }
 
+  private[models] def parseTargets(jsonTargets: Json): List[Target] = {
+    def parseTarget(key: String, value: String): Option[Target] = {
+      key.toLowerCase match {
+        case "stack" => Some(Stack(value))
```

Comment:
I think this might confuse people because these tags are all (strangely) fussy about their case elsewhere. By convention we use `Stack`, `App` and `Stage`.

Reply:
Oh sorry, I missed the `toLowerCase`. Do we need to lower case it here, or can we just assume people will use the correct case? If you give people the chance to misconfigure something they will do it!

Reply:
`toLowerCase` felt like an easy win for catching a small frustration that could creep into the test data or prod data. Maybe we should error, yet I feel casing is an easy thing to recover from.

If the `config.json` is ultimately going to be machine generated, where should the responsibility of sanity checking it fall? I feel like the json generator should hold the majority of that responsibility... and maybe that allows anghammarad to mostly trust the input string?